mitodl/mitxpro
# fix: make vat_id blank only to enable profile update

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/3429, https://github.com/mitodl/hq/issues/3494

### Description (What does it do?)
We recently added `vat_id` in https://github.com/mitodl/mitxpro/pull/2764
We made it nullable and It breaks for the existing users who had their profile before this feat. And it they don't add a value for the vat_id in their profile the `Continue` button doesn't work.
It also made it difficult to reproduce the issue because the vat_id field is hidden by default.
So, When those users update their profile their `Continue` button seems like not working but actually, It's adding an error in the `Vat` field that is hidden so the error message is not visible.

This is because for exiting users, the `vat_id` is `null` instead of `""` an empty string. So, the frontend validation breaks.

The error that's populated is:

<img width="784" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/300ef5b8-e954-4ef2-a26f-380ed7da034d">

Check `null` vat_id API Response:
<img width="526" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/36bf82da-68aa-482c-862f-848df77e2157">


### How can this be tested?
1. Create some accounts (Preferable if you have existing accounts before this https://github.com/mitodl/mitxpro/pull/2764)
2. If point#1 is not possible, you need to replicate the API response to return `null` in the `vat_id` field.
3. To make the `vat_id` null, You can use these steps:

```python
1. docker-compose exec web bash
2. ./manage.py dbshell
3. Find the profile you want to test with and run SQL:

update users_legaladdress set vat_id=NULL where id=3;

5. Check that you get `null` in the response from `/me` API for the vat_id
``` 
4. Once reproduce, get back to this branch and run migrations
6. Now test, the profile update should work fine and there should be no `null` values for `vat_id` in the DB.
